lib/cocoapods-core/specification/dsl.rb

```diff
@@ -1047,7 +1047,7 @@ def dependency=(args)
       SCRIPT_PHASE_OPTIONAL_KEYS = [:shell_path, :input_files, :output_files, :input_file_lists, :output_file_lists,
                                     :show_env_vars_in_log, :execution_position, :dependency_file].freeze
 
-      EXECUTION_POSITION_KEYS = [:before_compile, :after_compile, :any].freeze
+      EXECUTION_POSITION_KEYS = [:before_compile, :after_compile, :before_headers, :after_headers, :any].freeze
```

Comment:
if these are added and serialized and older version of cocoapods will not be able to parse them, I think that is OK and the author should use `s.cocoapods_version` dsl to specify >=. 1.11.0.

Reply:
I wonder if we could expose that in the docs somehow to clarify that if you use those keys you need X version of CocoaPods